skglm/solvers/group_bcd_solver.py

```diff
@@ -0,0 +1,86 @@
+import numpy as np
+from numba import njit
+
+
+def bcd_solver(X, y, datafit, penalty, w_init=None,
+               max_iter=1000, tol=1e-7, verbose=False):
+    """Run a group BCD solver.
+
+    Parameters
+    ----------
+    X : array, shape (n_samples, n_features)
+        Design matrix.
+
+    y : array, shape (n_samples,)
+        Target vector.
+
+    datafit : QuadraticGroup
+        DataFit object.
+
+    penalty : WeightedGroupL1
```

Comment:
`instance of WeightedGroupL1`

Reply:
Btw why should it be such an instance and not a general Penalty?

Reply:
you're right: `instance of Penalty`